ql/src/java/org/apache/hadoop/hive/ql/parse/repl/dump/events/AlterTableHandler.java

```diff
@@ -82,14 +92,71 @@ private Scenario scenarioType(org.apache.hadoop.hive.metastore.api.Table before,
     }
   }
 
+  // return true, if event needs to be dumped, else return false.
+  private boolean handleForTableLevelReplication(Context withinContext) {
+    String oldName = before.getTableName();
+    String newName = after.getTableName();
+
+    if (ReplUtils.tableIncludedInReplScope(withinContext.replScope, oldName)) {
+      // If the table is renamed after being added to the list of tables to be bootstrapped, then remove it from the
+      // list of tables to be bootstrapped.
+      boolean oldTableIsPresent = withinContext.removeFromListOfTablesForBootstrap(before.getTableName());
```

Comment:
Let's say, a non-acid table is converted to external. Now, since, both names match ReplScope, this event is dumped. Now, all the other events on this alter table would be skipped as BOOTSTRAP_EXTERNAL config is true. Now, we replicate the event and also, bootstrap the table which may fail. This is not related to this patch but still want to check if it can happen. I noticed, we block change table type only if STRICT_MANAGED=true.

Reply:
bootstrap external or bootstrap acid is handled in shuldreplicate method ..the control should not reach till here

Reply:
I think, we look at beforeTableObject in case of shouldReplicate method which allows this event as the before table was not external or ACID.

Reply:
i think new Jira is required for this ..willl create one ..thus scenario needs some more thinking